# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1509224889 -7200
#      Sat Oct 28 23:08:09 2017 +0200
# Node ID cbb3e02244a4785f490a938380a5bb250507d5ea
# Parent  abf23a47936922bdeb9af8f776589f4fdf3375b7
Fix regression introduced by cset 78ff6ba5cbb1 (bug #52293).

* gl-render.cc (openglrenderer::draw_surface): fix color indexing

diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -2380,17 +2380,17 @@ namespace octave
 
     if (! props.facecolor_is ("none"))
       {
         if (fa_mode == 0)
           {
             fa = props.get_facealpha_double ();
             if (fc_mode == UNIFORM || fc_mode == TEXTURE)
               {
-                glColor4d (fcolor(1), fcolor(2), fcolor(3), fa);
+                glColor4d (fcolor(0), fcolor(1), fcolor(2), fa);
                 if (fl_mode > 0)
                   {
                     for (int i = 0; i < 3; i++)
                       cb[i] = as * fcolor(i);
                     glMaterialfv (LIGHT_MODE, GL_AMBIENT, cb);
 
                     for (int i = 0; i < 3; i++)
                       cb[i] = ds * fcolor(i);
